cmd/infra/ibmcloud_powervs/create.go

```diff
@@ -396,109 +356,87 @@ func (infra *Infra) createCloudInstance(option *CreateInfraOptions) (resourceIns
 		return nil, fmt.Errorf("create cloud instance returned nil")
 	}
 
-	if *resourceInstance.State == "active" {
+	if *resourceInstance.State == activeStateL {
 		return resourceInstance, nil
 	}
 
-	retryExceeds := false
-	ttl := make(chan bool, 1)
-	go func() {
-		for {
-			if retryExceeds {
-				break
-			}
-			resourceInstance, _, err = rcv2.GetResourceInstance(&resourcecontrollerv2.GetResourceInstanceOptions{ID: resourceInstance.ID})
-			log.Log.Info("Waiting for cloud instance to up", "id", resourceInstance.ID, "state", *resourceInstance.State)
-			if *resourceInstance.State == "active" || err != nil {
-				break
-			}
-			time.Sleep(time.Second)
+	f := func() (bool, error) {
+		resourceInstance, _, err = rcv2.GetResourceInstance(&resourcecontrollerv2.GetResourceInstanceOptions{ID: resourceInstance.ID})
+		log.Log.Info("Waiting for cloud instance to up", "id", resourceInstance.ID, "state", *resourceInstance.State)
+
+		if err != nil {
+			return false, err
 		}
-		ttl <- true
-	}()
 
-	select {
-	case <-time.After(10 * time.Minute):
-		retryExceeds = true
-		return nil, fmt.Errorf("time exceeds waiting for cloud instance to up")
-	case <-ttl:
+		if *resourceInstance.State == activeStateL {
+			return true, nil
+		}
+		return false, nil
 	}
 
+	err = wait.PollImmediate(time.Second*5, time.Minute*5, f)
```

Comment:
wondering if these waiting and pooling interval can be defined as const for better reading

Reply:
is 5mins enough?

Reply:
this interval and timeout differs for diff resource
currently interval is 5 second in all resources monitoring but timeout is different, 5 mins for vpc and cloud instance to be up after creation and for cloud connection and dhcp server using 30 mins timeout.
will make them as constant.

Reply:
make sense.!

Reply:
made these durations as const.